### ISSUE TITLE
Remove boost::program options

### ISSUE THREAD

**Issue by aeantipov** [aeantipov/pomerol#33]
Boost::program_options is incompatible with old gcc, clang and builds fail

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=797917

**Comment by krivenko** [comment]
I checked three command-line parsing header-only libraries.

* [TCLAP](https://github.com/mirror/tclap)
* [args](https://github.com/Taywee/args)
* [cxxopts](https://github.com/jarro2783/cxxopts)

All three come with the same caveat: There is no way to simulate Boost's [multitoken()](https://www.boost.org/doc/libs/1_68_0/doc/html/boost/program_options/typed_value.html#id-1_3_31_9_10_1_1_7_5_8-bb) feature, which allows us to write
```
anderson.pomerol --levels 1.0 2.0 --hoppings 3.0 4.0
```
With those libraries, it has to be either
```
anderson.pomerol --levels 1.0,2.0 --hoppings 3.0,4.0
```
or
```
anderson.pomerol --levels "1.0 2.0" --hoppings "3.0 4.0"
```

@aeantipov How much of a problem do you think this change of the command line format would be?

**Comment by aeantipov** [comment]
I think it's fine or actually it's even better when we need to provide arguments via a comma. 

cxxopts [support std::vector<T> arguments](https://github.com/jarro2783/cxxopts#stdvectort-values), so something like that would work.

**Comment by krivenko** [comment]
cxxopts looked like a perfect choice as its API is very similar to that of boost::program_options. So I tried it and got disappointed.

* For some reason, dots are disallowed in option names, and we do need them (`gf.eta`, `gf.step`, etc).
* An option with a single-character name (for instance, `--U`) is always assumed to be a short option and has to be used as `-U` (https://github.com/jarro2783/cxxopts/issues/249).
* Not really a problem, but a rather strange decision to define default values in their unparsed form, i.e. as strings.

I have some positive experience using *args* and will try it next, but it will require some redesign of the programs.

**Comment by krivenko** [comment]
Fixed by #40.
